cosima_cookbook/querying.py

```diff
@@ -7,28 +7,51 @@
 import logging
 import os.path
 import pandas as pd
-from sqlalchemy import func, distinct
+from sqlalchemy import func, distinct, select, or_
 import warnings
 import xarray as xr
 
 from . import database
-from .database import NCExperiment, NCFile, CFVariable, NCVar
+from .database import NCExperiment, NCFile, CFVariable, NCVar, Keyword
 
 
 class VariableNotFoundError(Exception):
     pass
 
-def get_experiments(session):
+def get_experiments(session, experiment=True, keywords=None, all=False, **kwargs):
     """
     Returns a DataFrame of all experiments and the number of netCDF4 files contained 
     within each experiment.
     """
 
-    q = (session
-         .query(NCExperiment.experiment,
-                func.count(NCFile.experiment_id).label('ncfiles'))
-         .join(NCFile.experiment)
-         .group_by(NCFile.experiment_id))
+    # Determine which attributes to return. Special case experiment
+    # as this is the only one that defaults to True
+    columns = []
+    if experiment:
+        columns.append(NCExperiment.experiment)
+
+    for f in NCExperiment.metadata_keys + ['root_dir']:
+        # Explicitly don't support returning keyword metadata
+        if f == 'keywords': continue
+        if kwargs.get(f, all):
+            columns.append(getattr(NCExperiment, f))
+
+    if keywords is not None:
+
+        if type(keywords) == str:
+            keywords = [ keywords ]
+
+        q = (session
+            .query(*columns, func.count(NCFile.ncfile).label('ncfiles'))
+            .join(NCFile.experiment)
+            .filter(NCExperiment.keywords.in_(keywords))
```

Comment:
Maybe to get the _and_ behaviour:

```suggestion
            .filter(*(NCExperiment.keywords == k for k in keywords))
```

Reply:
Sweet! Yeah that works a treat. Thanks.